### DISCUSSION TITLE
On Screen Notes

### DISCUSSION THREAD

**Discussion by Phobosmarx** [runelite/runelite#16709] [Ideas]
Similar to Screen Markers, I would really like to be able to write down on screen some notes and have them stay there, at a reasonable size. Screen Markers allows you to label your markers but the text is so small its basically illegible. This should be a relitively easy plugin to make as you could adapt existing to focus on Text rather than Markers

**Comment by neeerp** [comment]
I'll look into trying to make a plugin for this over the next few days!

**Reply by Phobosmarx** [reply]
What a legend!

**Reply by SL90Battleship** [reply]
+1 for this

**Reply by neeerp** [reply]
Hey, I haven't had as much time as I'd have hoped and I'm also not entirely sure what I'm doing, so 'over the next few days' may have been a little optimistic. At this point, all I've really done is play around with making my own side panel (LOL).

Anyway, I should probably make sure I understand what everyone's looking for here... With the existing screen marker plugin, when you create a new marker you drag and drop this little box (see screenshot). Is what you're looking for the ability to type text into that box and adjust the font size/color/etc, and have that otherwise behave like a typical marker?
![newmarker](https://github.com/runelite/runelite/assets/22756295/0862a70e-8396-4d8b-bfd9-3e3a84ce3e92)

**Reply by SL90Battleship** [reply]
I can't speak for the original poster, but the description of how it works with size changes on the box and the change on size for the text itself with an easy ability to edit. 
Me personally, I want to be able to write a note of X/1400 collection logs and then update it when it changes. I don't want anything fancy to auto do it but I would like to be able to write it on screen myself!

**Reply by neeerp** [reply]
I figured out how to properly render text that wraps/stays within a box:

![notes](https://github.com/runelite/runelite/assets/22756295/3d0e3678-d5dc-4fdd-b527-52da55acf7a8)

Sorry about the slow progress on this. Can't say it'll get much faster as I've got quite a lot going on right now :/

**Reply by geheur** [reply]
Why not just make an overlay that shows text, like e.g. the clue scroll overlay? These can be dragged and resized like other overlays by holding alt and dragging, and automatically wrap text.

**Reply by neeerp** [reply]
Ah, I hadn't even thought to look at that, nor did I realize you could drag stuff around (actually, I personally can't because my alt key is bound by my window manager... lol). 

Thanks for pointing that out, I'll look into how those are implemented.

**Reply by AvariceBrah** [reply]
Any chance of freshening it up a wee bit just to ensure all is good?

**Reply by neeerp** [reply]
I don't think I ever got around to finishing this. I could maybe take a crack at vibe coding this later

**Reply by AvariceBrah** [reply]
My buddy found one named on-screen notes so idk if needed but I'll go ahead and star the repo in case ya do

**Reply by roskabytes** [reply]
@AvariceBrah I haven't ever been able to get it to work. Have you had any luck? The dev hasn't responded to any requests since mid 2024 judging by the repo so I assumed it's abandoned

**Reply by raiyni** [reply]
this plugin exists https://runelite.net/plugin-hub/show/on-screen-notes
